policy_packs/aws/ssm/enforce_install_packages_on_ec2_instances_using_ssm_package_installer/README.md

```diff
@@ -0,0 +1,198 @@
+---
+categories: ["security"]
+primary_category: "security"
+---
+
+# Enforce Install Packages on AWS EC2 Instances Using SSM Package Installer
+
+Enforcing the installation of packages on AWS EC2 instances using the SSM Package Installer is crucial for maintaining a consistent, secure, and automated software deployment process. This measure ensures that all package installations are managed and logged through AWS Systems Manager, enhancing security, compliance, and operational efficiency by providing centralized control and monitoring of software installations.
+
+This [policy pack](https://turbot.com/guardrails/docs/concepts/resources/smart-folders) is designed to allow package installation on instances with or without direct access to original package source and can help you configure the following settings for SSM documents:
+
+- Set the region ARN where the SSM resources will be deployed
+- Set the IAM role ARN to be used by SSM to run documents
+- Create three SSM documents in the specified region
+  - Turbot-LinuxPackageInstaller-PublicNetwork
+  - Turbot-LinuxPackageInstaller-PrivateNetwork
+  - Turbot-LinuxS3PackageInstaller
+
+## Documentation
+
+- **[Review Policy settings →](https://hub-guardrails-turbot-com-git-development-turbot.vercel.app/policy-packs/enforce_install_packages_on_ec2_instances_using_ssm_package_installer/settings)**
+
+## Getting Started
+
+### Requirements
+
+- [Terraform](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli)
+- Guardrails mods:
+  - [@turbot/aws-ssm](https://hub-guardrails-turbot-com-git-development-turbot.vercel.app/aws/mods/aws-ssm)
+- AWS resources to run the stack control:
+  - S3 bucket to store packages
+  - An IAM role for SSM document execution with the following configured permissions:
+
+    ```json
+    {
+      "Version": "2012-10-17",
+      "Statement": [
+        {
+          "Sid": "SSMDocumentExecutionPermissions",
+          "Effect": "Allow",
+          "Action": [
+            "ssm:DescribeInstanceInformation",
+            "ssm:ListCommandInvocations",
+            "ssm:ListCommands",
+            "ssm:SendCommand",
+            "s3:PutObject"
+          ],
+          "Resource": "*"
+        }
+      ]
+    }
+    ```
+
+  - An IAM role for the EC2 instance with the following configured permissions:
+
+    ```json
+    {
+      "Version": "2012-10-17",
+      "Statement": [
+        {
+          "Sid": "EC2InstancePermissions",
+          "Effect": "Allow",
+          "Action": [
+            "ssm:DescribeInstanceInformation",
+            "ssm:ListCommandInvocations",
+            "ssm:ListCommands",
+            "ssm:SendCommand",
+            "s3:PutObject"
+          ],
+          "Resource": "*"
+        }
+      ]
+    }
+    ```
+
+### Using the SSM Documents
```

Comment:
I'd suggest moving this at the bottom of the README since the documents are created once the policy pack is put into enforce mode.